### ISSUE TITLE
Modal support of icon

### ISSUE THREAD

**Issue by pranaykute06** [iTwin/iTwinUI#586]
#### Feature
Add support for icon in the modal.

User can provide the required icon if they want to.

such modal can be used for delete confirmation etc....

#### Examples
![image](https://user-images.githubusercontent.com/82197220/162974880-4daaa9d2-68ee-4a5e-947e-10c3eb8f69ae.png)

![image](https://user-images.githubusercontent.com/82197220/162974904-6be3fa47-94ac-4327-af8e-79e45b7f5960.png)

**Comment by mayank99** [comment]
Hi @pranaykute06, thanks for the issue.

This would be a(nother) great use case for a generic "text with icon" component. I don't think we would want to add any modal-specific CSS for this.

@FlyersPh9 What is the correct spacing that should be applied here?

**Comment by FlyersPh9** [comment]
Areas that I think could make use of the icon + text + right aligned stuff that I can think of:
- Information panel title bar
- Modal title bar
- Expandable block title bar
- Menu items

I would probably replicate what is present in information panel, which is an 8px gap using `$iui-s`.
![Screen Shot 2022-04-12 at 5 43 22 PM](https://user-images.githubusercontent.com/849817/163059424-1fe9ba4a-a054-4b55-b918-00e2fc634834.png)

**Comment by FlyersPh9** [comment]
A mixin like this could be reused in a lot of the places I mentioned in my last comment:
```scss
@mixin iui-title-bar($title-bar-label-font-size: $iui-font-size-subheading, $title-bar-sublabel-font-size: $iui-font-size-small) {
  justify-content: space-between;

  &,
  &-left,
  &-right {
    display: flex;
    align-items: center;
  }

  &-left,
  &-right {
    gap: $iui-s;
  }

  &-icon {
    display: flex;
    width: $iui-icons-default;
    height: $iui-icons-default;
    @include themed {
      fill: t(iui-icons-color);
    }
  }

  &-title {
    display: flex;
    flex-direction: column;
    flex: auto;
    min-width: 0;
    @include themed {
      color: t(iui-color-foreground-body);
    }

    &-label,
    &-sublabel {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    &-label {
      font-size: $title-bar-label-font-size;
      @include themed {
        opacity: t(iui-opacity-2);
      }
    }

    &-sublabel {
      font-size: $title-bar-sublabel-font-size;
      @include themed {
        opacity: t(iui-opacity-3);
      }
    }
  }
}
```

**Comment by mayank99** [comment]
> A mixin like this could be reused in a lot of the places I mentioned in my last comment:

Mixin is still too low level imo (it's an implementation details that gets "compiled" away). We should offer a simple class and component that the user can place anywhere. Nothing fancy, just this should be enough:

```css
.iui-text-with-icon {
  display: flex;
  align-items: center;
  gap: $iui-s;

  &-icon {
    @include iui-icons-default;
    flex-shrink: 0;
  }

  &-text {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
```

We already have classes for `iui-icons-default` available to the user, which we might even want to expose as a component in react. As for sublabel support, that can come in the future as it's not relevant to this issue.

**Comment by FlyersPh9** [comment]
👆 this suggestion makes sense to me.  A component-agnostic set of classes that can be reused within multiple components.  I believe this CSS would belong within `/utils/`.  The end result should be a set of elements / classes that the developer could put inside [`.iui-dialog-title-bar`](https://github.com/iTwin/iTwinUI/blob/8db64436470527571eae321272ac1cf1555e13c9/packages/itwinui-css/backstop/tests/dialog.html#L142).  This should also be usable in other places, but for initial PR let's make sure it works for dialog.

**Comment by mayank99** [comment]
I should add that I think my suggested code is still doing too much and not enough at the same time. I don't think the text overflow and icon sizing belong in this component, it should only handle spacing.

For icons, I think we would want another component which accepts any svg. This can handle sizing and fill.
```tsx
<Icon svg={<SvgInfoCircular />} />
```

And then it can be passed to the new utility component.
```tsx
<TextWithIcon>
  <Icon svg={<SvgInfoCircular />} />
  <span>Some text here</span>
</TextWithicon>
```

**Comment by FlyersPh9** [comment]
Might be addressed in #829.

**Comment by mayank99** [comment]
This will be possible in the next release using `Flex` with `Icon`.

```jsx
const MyDialogTitle = () => (
  <Flex>
    <Icon aria-hidden><SvgInfoCircular /></Icon>
    This is the title
  </Flex>
)
```

```jsx
<Dialog>
  <Dialog.Main>
    <Dialog.TitleBar titleText={<MyDialogTitle />} />
    ...
  </Dialog.Main>
  ...
</Dialog>
```
